src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/EJSONWrapper.java

```diff
@@ -0,0 +1,86 @@
+package io.stargate.sgv2.jsonapi.api.model.command.clause.filter;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+
+/**
+ * Intermediate type for further processing: used with {@link JsonLiteral} and {@link
+ * JsonType#EJSON_WRAPPER} to store special types in JSON (e.g. binary data). Currently only one
+ * type is supported but this can be extended in the future.
+ */
+public class EJSONWrapper {
+  // Actual DRY keys
+  public static final String $BINARY = "$binary";
+
+  public enum EJSONType {
+    BINARY($BINARY);
+
+    private final String key;
+
+    EJSONType(String key) {
+      this.key = key;
+    }
+
+    public static EJSONType fromKey(String key) {
+      // With very few entries just do this: can build lookup Map if needed
+      // in future
+      return switch (key) {
+        case $BINARY -> BINARY;
+        default -> null;
+      };
+    }
+
+    public String key() {
+      return key;
+    }
+  }
+
+  private final EJSONType type;
+  private final JsonNode value;
+
+  public EJSONWrapper(EJSONType type, JsonNode value) {
+    this.type = Objects.requireNonNull(type);
+    this.value = Objects.requireNonNull(value);
```

Comment:
I did not add this due to there possible being other EJSON types (all with their own constraints), but now that you mention this... perhaps. Let me think about this.

Reply:
Oh wait, actually, I am not sure we can do this. There is (and I think, needs to be) dualism with `EJSONWrapper`:

1. When Inserting (json->cql), we get textual `JsonNode` from JSON doc (since there is no native binary type), and will defer decoding until we know 100% sure we are to convert to Blob type. So we have to allow `TextNode` on this direction
2. When building response (cql->json) we will create wrapper with `BinaryNode` (or if we wanted to, could create pre-encoded Base64 wrap as `TextNode`

So ultimately logic is bit complicated and I don't think it makes sense to add rigid assertion here. Our tests need to validate correct functioning end-to-end. Internal state should remain somewhat flexible I think.

Reply:
sounds good